### PR TITLE
App frameworking

### PR DESCRIPTION
**Pull request by ryno83**
For the enhancement/issue #25.
This is a new implementation on basis of AppFramework (ask by @DeepDiver1975). What do you think about it ? I don't think it's finished but I'm asking you some feedbacks and comments to see if I'm on the right way !

Thank you guys

### CODE REVIEW

File: l10n/en_GB.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ OC.L10N.register(
     "Connect your Contacts" : "Connect your Contacts",
     "Access files via WebDAV" : "Access files via WebDAV",
     "Documentation" : "Documentation",
+	"Your full name has been changed" : "Your full name has been changed",
```
Comment (DeepDiver1975) [inline l10n/en_GB.js]: Please remove any changes to the translation files - they will be generated

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+dist: clean
+	mkdir -p $(source_dir)
+	tar cvzf $(source_dir)/$(package_name).tar.gz $(project_dir) \
+	--exclude-vcs \
+	--exclude=$(project_dir)/build/artifacts \
+	--exclude=$(project_dir)/js/node_modules \
+	--exclude=$(project_dir)/js/coverage
+
+appstore: appstore_package
```
Comment (DeepDiver1975) [inline Makefile]: from my understanding appstore related stuff can be removed - this is a shipped app as part of the release and not available on the appstore

File: README.md [hunk 3]
```diff
@@ -1,6 +1,17 @@
...
 
-A first run wizard that explains the usage of ownCloud to new users
+## Publish to App Store
 
-[![Build Status](https://travis-ci.org/owncloud/firstrunwizard.svg?branch=master)](https://travis-ci.org/owncloud/firstrunwizard)
+First get an account for the [App Store](http://apps.owncloud.com/) then run:
+
+    make appstore
+
+**ocdev** will ask for your App Store credentials and save them to ~/.ocdevrc which is created afterwards for reuse.
```
Comment (DeepDiver1975) [inline README.md]: no need to have this generated readme here

File: appinfo/app.php [hunk 4]
```diff
@@ -1,35 +1,22 @@
...
- * 
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
- *  
- * You should have received a copy of the GNU Affero General Public
- * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
- * 
+ * @author Your Name <mail@example.com>
```
Comment (DeepDiver1975) [inline appinfo/app.php]: name, email is missing

File: appinfo/app.php [hunk 5]
```diff
@@ -1,35 +1,22 @@
...
-OCP\Util::addStyle( 'firstrunwizard', 'colorbox');
-OCP\Util::addScript( 'firstrunwizard', 'jquery.colorbox');
-OCP\Util::addScript( 'firstrunwizard', 'firstrunwizard');
+if (\OCP\User::isLoggedIn()) {
+	$config = \OC::$server->getConfig();
+	$user = \OC::$server->getUserSession()->getUser();
 
-OCP\Util::addStyle('firstrunwizard', 'firstrunwizard');
+	if ($config->getUserValue($user->getUID() , 'firstrunwizard' , 'show' , '1' ) === '1'){
+		header('Location: ' . \OC::$server->getURLGenerator()->linkToRoute('firstrunwizard.page.index'));
```
Comment (DeepDiver1975) [inline appinfo/app.php]: direct header call in app.php is dangerous as it prevents and other apps to execute their app.php :-1:

File: appinfo/routes.php [hunk 6]
```diff
@@ -1,17 +1,27 @@
...
- * See the COPYING-README file.
+ * later. See the COPYING file.
+ *
+ * @author Your Name <mail@example.com>
+ * @copyright Your Name 2015
  */
 
-/** @var $this OC\Route\Router */
-
-$this->create('firstrunwizard_enable', 'ajax/enable.php')
```
Comment (DeepDiver1975) [inline appinfo/routes.php]: looks like the explicit enable/disable feature has been removed??

File: appinfo/app.php (hunk 4 above)
Comment (ryno83) [inline appinfo/app.php]: Should I put back the lines 6 to 21 ?

**Comment by ryno83** [comment]
@DeepDiver1975 is it a better way to load the modal box : 2373a66 ? It's very similar to the actual firstrunwizard.

**Comment by ryno83** [comment]
@DeepDiver1975 Maybe i'm wrong but if I look at the code of the current firstrunwizard, the enable feature is not use at all and the disable is use when closing the modal box. 

I'm disabling the app for the user immediatly after open.js was loaded by app.php. This prevent the modal box from loading again and again.

**Comment by jancborchardt** [comment]
@DeepDiver1975 @karlitschek @Raydiation can you review this again?

**Comment by karlitschek** [comment]
makes sense :+1:

**Comment by ryno83** [comment]
@jancborchardt  Could you tell me why github say that this branch is having conflict ?

**Comment by jancborchardt** [comment]
@ryno83 because it has changes which conflict with current master. You need to rebase your branch like so:
```
git checkout master
git pull
git checkout <your branch name>
git rebase master
```
And then fix the conflicts, and then push them here. Maybe you need to force push: `git push -f origin <your branch name>`

**Comment by ryno83** [comment]
@jancborchardt The conflict is still here even after fixing the conflict (travis.yml and build.xml) ! Is it because on my fork I have AppFrameworking and appframeworking (lowercase) branches ? On my computer I don't see the lowercase one.

**Comment by jancborchardt** [comment]
I’m not sure as there are really a _ton_ of changes.

@DeepDiver1975 @karlitschek can you advise here?

**Comment by DeepDiver1975** [comment]
TBH: this is a big change and we have the usual problem that too many thing have been put into one PR.

I suggest to have one PR which eliminates */ajax/* and introduces a controller.
Anything else can be done in a later step.

THX

**Comment by DeepDiver1975** [comment]
moving to 9.0 - we are far beyond feature freeze of 8.2

File: appinfo/info.xml [hunk 7]
```diff
@@ -1,14 +1,15 @@
...
-	<default_enable/>
-	<ocsid>166055</ocsid>
-</info>
+    <id>firstrunwizard</id>
+    <name>First Run Wizard</name>
+    <description>This application enables a pop-up window when a user first logs into ownCloud. This window welcomes new users to ownCloud, and contains links to the standard ownCloud desktop client, Google Play store app, and the Apple App Store app. The window also shows a link for accessing ownCloud via webdav. At any time, a user can click on a first run wizard button in their personal page to cause the app to reappear.
+The First Run Wizard can be customized to meet specific design goals, or to change links and messages. Create an appropriate theme following the Theming documentation, and changes to the look and feel of this Wizard are simple. More information is available in the First Run Wizard documentation, as well as the theming documentation.</description>
+    <licence>AGPL</licence>
+    <author>Frank Karlitschek, Jan-Christoph Borchardt, Renaud Fortier</author>
+    <version>1.0.1</version>
```
Comment (nickvergessen) [inline appinfo/info.xml]: version must not be lower then the currently used one.
Should be 1.1.1 at least!
:boom: :bomb: :x:

**Comment by ryno83** [comment]
I close this pull request because I created a new one #32 with less modification. Please review

**Pull request closed without merging**